llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -3231,9 +3231,8 @@ void InstCombinerImpl::handleUnreachableFrom(
     MadeIRChange = true;
   }
 
-  // RemoveDIs: to match behaviour in dbg.value mode, drop debug-info on
-  // terminator too.
-  BB->getTerminator()->dropDbgValues();
+  if (!DT.isReachableFromEntry(BB))
```

Comment:
Your previous version here was correct. handleUnreachableFrom already guarantees that the code is unreachable. Checking isReachableFromEntry does not work as desired inside InstCombine (it does not make use of reachability knowledge private to InstCombine).